CMakeLists.txt

```diff
@@ -9,20 +9,16 @@ project(
 # ---- Warning guard ----
 
 # Protect dependents from this project's warnings if the guard isn't disabled
-set(example_warning_guard SYSTEM)
-if(example_INCLUDE_WITHOUT_SYSTEM)
-  set(example_warning_guard "")
+option(example_INCLUDE_WITHOUT_SYSTEM "Use SYSTEM keyword for include" OFF)
```

Comment:
This unnecessarily introduces a cache variable when a project is vendored. When left as is, a user can use either a cache variable or a normal variable from a superbuild CML. See [here](https://github.com/friendlyanon/cxx-static-shared-example/blob/master/all/CMakeLists.txt#L5) (although I decided to unnecessarily make it a cache variable there, oh well)

Reply:
I was under the impression, that using options to provide user configurable switches is the way to go 🤔 Its a nice way to show the user "here you go, these are the variables you may fiddle with". Figuring that out for example_INCLUDE_WITHOUT_SYSTEM difficult, it would need me to dig through your whole CML, check all if conditions, and then decide, if want to mess with it or not.

Reply:
This option is really only relevant when the code is vendored. If the library is being built directly, then this switch is not useful.  
The idea here is to mimic imported targets, as those provide implicitly system includes and when vendoring code, you don't get imported targets. It's a really small amount of code that by default provides a similar experience to consuming real CMake packages for the people who use `FetchContent` or other means to (in)directly `add_subdirectory` a dependency. This is also one of the reasons you should have alias targets (other reason being that targets with `::` won't turn into `-ltypo_in_naem`, but a hard error).  
For those that wish/are required to observe warnings from dependencies, it is still possible without patching to have that. I admit that without people knowing about this it might not be so obvious, but I try my best by having this is high up in the code as possible with a comment.